sdk/Pulumi/Deployment/Deployment_Prepare.cs

```diff
@@ -166,60 +166,10 @@ private static async Task<Pulumirpc.Callback> AllocateTransform(Callbacks callba
-                    {
-                        // Else need to wrap it in an Output<T>
-                        var elementType = typeof(object);
-                        if (outputData.Value != null)
-                        {
-                            elementType = outputData.Value.GetType();
-                        }
-                        var outputDataType = typeof(Pulumi.Serialization.OutputData<>).MakeGenericType(elementType);
-                        var createOutputData = outputDataType.GetConstructor(new[]
-                        {
-                            typeof(ImmutableHashSet<Resource>),
-                            elementType,
-                            typeof(bool),
-                            typeof(bool)
-                        });
-                        if (createOutputData == null)
-                        {
-                            throw new InvalidOperationException(
-                                $"Could not find constructor for type OutputData<T> with parameters " +
-                                $"{nameof(ImmutableHashSet<Resource>)}, {elementType.Name}, bool, bool");
-                        }
-
-                        var typedOutputData = createOutputData.Invoke(new object?[] {
-                            ImmutableHashSet<Resource>.Empty,
-                            outputData.Value,
-                            outputData.IsKnown,
-                            outputData.IsSecret });
-
-                        var createOutputMethod =
-                            typeof(Output<>)
-                                .MakeGenericType(elementType)
-                                .GetConstructors(BindingFlags.NonPublic | BindingFlags.Instance)
-                                .First(ctor =>
-                                {
-                                    // expected parameter type == Task<OutputData<T>>
-                                    var parameters = ctor.GetParameters();
-                                    return parameters.Length == 1 &&
-                                           parameters[0].ParameterType == typeof(Task<>).MakeGenericType(outputDataType);
-                                })!;
-
-                        var fromResultMethod =
-                            typeof(Task)
-                                .GetMethod("FromResult")!
-                                .MakeGenericMethod(outputDataType)!;
-
-                        var outputObject = createOutputMethod.Invoke(new[] { fromResultMethod.Invoke(null, new[] { typedOutputData }) });
-                        props.Add(kv.Key, outputObject);
-                    }
+                    // The data from the engine will include output values, which will be deserialized
+                    // into instances of `Output<T>`, which will track whether the value is known,
+                    // is secret, and dependencies. We can ignore those values on outputData.
+                    props.Add(kv.Key, outputData.Value);
```

Comment:
Doesn't this still need to check the OutputData for known/secretness? Doesn't look like Deserialise is returning Output<T> for those, just for actual output wire objects.

Reply:
I don't think so. Shouldn't everything be coming across from the engine to transforms as output values? In which case, those will already have the known/secretness/dependency tracking.

Reply:
For transforms yes, but isn't this used for other deserialisation for invokes/resources?

Reply:
Ah no ignore me, this is specific to transforms. Should be fine then.